api/METRICS.md

```diff
@@ -35,72 +45,263 @@ acc_ratio = accuracy_score_ratio(y_true, y_pred, sensitive_features=sf, **other_
 acc_group_min = accuracy_score_group_min(y_true, y_pred, sensitive_features=sf, **other_kwargs)
 ```
 
-## Proposal
+## Proposal details
 
-*Function signatures*
+The items that are not implemented yet are marked as `[TODO]`.
+
+### Metrics engine
 
 ```python
 group_summary(metric, y_true, y_pred, *, sensitive_features, **other_kwargs)
 # return the group summary for the provided `metric`, where `metric` has the signature
 # metric(y_true, y_pred, **other_kwargs)
 
-make_metric_group_summary(metric)
+make_group_summary(metric) [TODO]
 # return a callable object <metric>_group_summary:
-# <metric>_group_summary(...) = group_summary(<metric>, ...)
+# <metric>_group_summary(...) = group_summary(metric, ...)
 
-# Transformation functions returning scalars
+# Transformation functions returning scalars (the definitions are below)

```

Comment:
Just wondering, have you also considered creating a summary class with methods rather than having separate functions for transformations? I.e. you would call `summary.difference()` rather than `difference_from_summary(summary)`, a bit like the pandas API. An advantage of this approach is that all available methods can be found at one place in the documentation and can be accessed easily in most IDEs. It also allows for shorter names. And you could create e.g. `summary.to_dataframe()` which can be convenient for more novice users who are not as familiar with transforming dictionaries to data frames.